### PR TITLE
fallback to current interpreter if matches at discovery

### PR DESCRIPTION
**Pull request by gaborbernat**
#994

### CODE REVIEW

**Comment by gaborbernat** [comment]
@asottile here it seems tox is not able to find pyton3.5 when running under python3.5 :open_mouth:

**Comment by asottile** [comment]
wow that seems pretty special -- that's windows if I'm reading correctly -- is the `py` launcher configured correctly? what happens with `py -3.5`

**Comment by gaborbernat** [comment]
```bash
2018-09-17T20:14:15.5777113Z py -3.5 -c "import sys; print(sys.executable)"
2018-09-17T20:14:15.6419862Z ##[command]"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\15c4da66-e82e-4a75-ab7d-6d7f0803f389.cmd""
2018-09-17T20:14:16.2417178Z Requested Python version (3.5) not installed
2018-09-17T20:14:16.4035016Z ##[error]Cmd.exe exited with code '103'.
```
fails with exit code 103 :thinking: definitely an Azure bug here.

**Comment by asottile** [comment]
whew!

**Comment by DavidStaheli** [comment]
Strange!  We'll look into this right now.  Thanks for the heads-up.

**Comment by gaborbernat** [comment]
@asottile the deliverable on our end I believe is in such cases before giving up we should check if the tox host executable matches up with the requested version, if so we should use it.

File: src/tox/interpreters.py [hunk 1]
```diff
@@ -130,6 +130,8 @@ def __str__(self):
 
     @tox.hookimpl
     def tox_get_python_executable(envconfig):
+        if envconfig.basepython == "python{}.{}".format(*sys.version_info[0:2]):
```
Comment (asottile) [inline src/tox/interpreters.py]: `[0:2]` is just `[:2]`

**Pull request merged**